prow/crier/reporters/gerrit/reporter.go

```diff
@@ -92,12 +93,21 @@ type JobReport struct {
 }
 
 // NewReporter returns a reporter client
-func NewReporter(cookiefilePath string, projects map[string][]string, lister ctrlruntimeclient.Reader) (*Client, error) {
+func NewReporter(cfg config.Getter, cookiefilePath string, projects map[string][]string, lister ctrlruntimeclient.Reader) (*Client, error) {
 	gc, err := client.NewClient(projects)
 	if err != nil {
 		return nil, err
 	}
 	gc.Authenticate(cookiefilePath, "")
+	go func() {
+		for {
+			orgReposConfig := cfg().Gerrit.OrgReposConfig
+			if orgReposConfig == nil {
+				continue
+			}
+			gc.UpdateClients(orgReposConfig.AllRepos())
```

Comment:
I think doing this constantly in a separate goroutine is still technically racy. We could alternatively make the gerrit client create new handlers for new repos when we find that it is missing from the same goroutine.

Reply:
both `UpdateClients` and `Authenticate` above are protected by the same lock, I'm not seeing any race condition here.

Reply:
The race I'm imagining could occur if the gerrit config is added at the same time a job that uses the config is added. Since the gerrit clients are added in a separate goroutine, we may try to handle the new job before the appropriate client has been loaded. This race is really unlikely, but unless creating the clients is expensive, it might be simpler and slightly less racy to grab a write lock and create the needed client if it is not available at the time of use without a background goroutine.  I don't think this is a big deal so feel free to keep this as is, just food for thought.